lib/logstash/inputs/http_poller.rb

```diff
@@ -133,11 +157,35 @@ def validate_request!(url_or_spec, request)
 
   public
   def run(queue)
-    @interval_thread = Thread.current
-    Stud.interval(@interval) do
-      run_once(queue)
+    @scheduler = Rufus::Scheduler.new(:max_work_threads => 1)
+    if @interval #interval option is checked first to give it higher precedence
```

Comment:
A `LogStash::ConfigurationError` should be raised if two schedule options are used by the user at the same time.

This could be DRYed up with code like:

```ruby
raise LogStash::ConfigurationError, "..." if @schedule.keys > 1
schedule_type = @schedule.keys.first
schedule_value = @schedule[schedule_type]
schedule_types = %w(cron every at in)
raise LogStash::ConfigurationError, "..." unless schedule_types.include?(schedule_type)

@scheduler.send(schedule_type, @schedule_value) { run_once(queue) }
```